src/build_function.jl

```diff
@@ -358,8 +358,8 @@ function numbered_expr(O::Symbolic,varnumbercache,args...;varordering = args[1],
                        lhsname=gensym("du"),rhsnames=[gensym("MTK") for i in 1:length(args)])
     O = value(O)
     if O isa Sym || isa(operation(O), Sym)
-        if haskey(varnumbercache,O)
-            (j,i) = varnumbercache[O]
+        (j,i) = get(varnumbercache, O, (nothing, nothing))
+        if !isnothing(j)
```

Comment:
This is very minor, but it's almost always better to use `j !== nothing` because `j` is not type-stable, so calling a generic function like `isnothing` requires multiple dispatch, while `===` immediately checks if `j` is singleton `nothing`.

Reply:
Good point, I hadn't thought about that!